client/search_v1alpha2.go

```diff
@@ -0,0 +1,40 @@
+// Copyright AGNTCY Contributors (https://github.com/agntcy)
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	searchtypes "github.com/agntcy/dir/api/search/v1alpha2"
+)
+
+func (c *Client) SearchV1alpha2(ctx context.Context, req *searchtypes.SearchRequest) (io.Reader, error) {
```

Comment:
i think it would be better to simply use versionless naming here (because we can always put these under version specific module under the client). and on the second part, id consider returning chans here
```suggestion
func (c *Client) Search(ctx context.Context, req *searchtypes.SearchRequest) (<-chan string, error) {
```